cmake/Modules/GeneratePkgConfig.cmake

```diff
@@ -146,10 +146,14 @@ function(generate_and_install_pkg_config_file _target _packageName)
 	set(_generate_target_dir "${CMAKE_CURRENT_BINARY_DIR}/${_target}-pkgconfig")
 	set(_pkg_config_file_template_filename "${_GeneratePkGConfigDir}/pkg-config.cmake.in")
 
-	# Avoid things in the CFLAGS -- which can come from INTERFACE_OPTIONS
-	# of dependencies -- that don't work with file(GENERATE). CMake bug 21074
-	string(REPLACE "<COMPILE_LANG_AND_ID:CUDA,NVIDIA>" "<COMPILE_LANGUAGE:CUDA>" _s "${_interface_compile_options}")
-	set(_interface_compile_options "${_s}")
+	# Since CMake 3.18 FindThreads may include a generator expression requiring a target, which gets propagated to us through INTERFACE_OPTIONS.
+	# Before CMake 3.19 there's no way to solve this in a general way, so we work around the specific case. See #4956 and CMake bug #21074.
+	if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.19)
+		set(target_arg TARGET ${_target})
```

Comment:
In this file all local variables (except loop variables) seem to be prefixed with underscore. Could you change `target_arg` to `_target_arg`, please?